test/client/fixtures/automation/click-test.js

```diff
@@ -1176,5 +1176,38 @@ $(document).ready(function () {
                     startNext();
                 });
         });
+
+        asyncTest('mouse or click events should not be raised if touch event was cancelled', function () {
+            const raisedEvents = [];
+
+            const events = [
+                'ontouchstart',
+                'ontouchend',
+                'ontouchmove',
+                'onmousedown',
+                'onmousemove',
+                'onmouseup',
+                'onclick'
+            ];
+
+            events.forEach(function (eventName) {
+                $el[0][eventName] = function (e) {
+                    if (eventName === 'ontouchstart')
+                        e.preventDefault();
+
+                    raisedEvents.push(eventName);
+                };
+            });
+
+            const click = new ClickAutomation($el[0], new ClickOptions());
+
+            click
+                .run()
+                .then(function () {
+                    deepEqual(raisedEvents, ['ontouchstart', 'ontouchend']);
+
+                    startNext();
+                });
+        });
```

Comment:
Please test the `outouchend` case as well. According to the code, if we prevent the `touchend` event the click will not fire.